security/pkg/k8s/controller/customdnsname.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"strings"
+
+	"istio.io/istio/pkg/log"
+)
+
+func ConstructCustomDNSNames(serviceAccounts []string, serviceNames []string,
+	namespace string, customDNSNames string) map[string]*DNSNameEntry {
+	result := make(map[string]*DNSNameEntry)
+	for i, svcAccount := range serviceAccounts {
+		result[svcAccount] = &DNSNameEntry{
+			ServiceName: serviceNames[i],
+			Namespace:   namespace,
+		}
+	}
+	if len(customDNSNames) > 0 {
+		customNames := strings.Split(customDNSNames, ",")
+		log.Infof("There are %v custom-defined DNS names", len(customNames))
```

Comment:
You can print them out using %v, if that's helpful.

Reply:
Done.